packages/@aws-cdk/aws-amplify/lib/app.ts

```diff
@@ -0,0 +1,347 @@
+import { IRole, LazyRole, ServicePrincipal } from '@aws-cdk/aws-iam';
+import { Construct, IResolvable, IResolveContext, IResource, Resource, SecretValue, Tag } from '@aws-cdk/core';
+import { CfnApp } from './amplify.generated';
+import { Branch, BranchBaseProps } from './branch';
+import { BuildSpec } from './buildspec';
+import { Domain, DomainBaseProps } from './domain';
+import { BasicAuthConfig, BasicAuthResolver, EnvironmentVariable, EnvironmentVariablesResolver } from './shared';
+
+/**
+ * App
+ */
+export class App extends Resource implements IApp {
+  /**
+   * Allow creating from App ID
+   *
+   * @param scope
+   * @param id
+   * @param appId
+   *
+   * @returns Stubbed App
+   */
+  public static fromAppId(scope: Construct, id: string, appId: string): IApp {
+    class Import extends Resource implements IApp {
+      public readonly appId = appId;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Unique Id for the Amplify App.
+   *
+   * @attribute
+   */
+  public readonly appId: string;
+
+  /**
+   * Name for the Amplify App.
+   *
+   * @attribute
+   */
+  public readonly appName: string;
+
+  /**
+   * ARN for the Amplify App.
+   *
+   * @attribute
+   */
+  public readonly appArn: string;
+
+  /**
+   * Default domain for the Amplify App.
+   *
+   * @attribute
+   */
+  public readonly appDefaultDomain: string;
+
+  /**
+   * Resource
+   */
+  private readonly resource: CfnApp;
+
+  private readonly basicAuthResolver: BasicAuthResolver = new BasicAuthResolver();
+
+  private readonly environmentVariablesResolver: EnvironmentVariablesResolver = new EnvironmentVariablesResolver();
+
+  private readonly customRulesResolver: CustomRulesResolver = new CustomRulesResolver();
+
+  constructor(scope: Construct, id: string, props: AppProps) {
+    super(scope, id, {
+      physicalName: props.appName
+    });
+
+    if (props.basicAuthConfig) {
+      this.addBasicAuth(props.basicAuthConfig.username, props.basicAuthConfig.password);
+    }
+
+    if (props.environmentVariables && props.environmentVariables.length > 0) {
+      this.environmentVariablesResolver.addEnvironmentVariables(...props.environmentVariables);
+    }
+
+    let roleArn: string | undefined;
+
+    if (props.serviceRole) {
+      roleArn = props.serviceRole.roleArn;
+    }
+
+    if (props.buildSpec && !props.serviceRole) {
+      const role = new LazyRole(scope, 'lazy-role', {
+        assumedBy: new ServicePrincipal('amplify.amazonaws.com')
+      });
+
+      roleArn = role.roleArn;
+    }
+
+    const resource = new CfnApp(this, 'App', {
+      accessToken: props.accessToken,
+      basicAuthConfig: this.basicAuthResolver,
+      buildSpec: (props.buildSpec) ? props.buildSpec.toBuildSpec() : undefined,
+      customRules: this.customRulesResolver,
+      description: props.description,
+      environmentVariables: this.environmentVariablesResolver,
+      iamServiceRole: roleArn,
+      name: props.appName,
```

Comment:
I think this needs to be `this.physicalName`

Reply:
Does it matter? appName is readonly and is the value used to set this.physicalName

Reply:
Yap. `this.physicalName` implements the support for `PhysicalName.GENERATE_IF_NEEDED`.